feat(components): add themeswitcher component, layout in app

Hellomyguest opened the pull request:
themeswitcher, styled by style prop, uses effector state, theme changes by configprovider
all content in app moved to ant-design layout

META DATA: issue13

Commit:
feat(components): add themeswitcher component, layout in app

themeswitcher, styled by css-in-js, uses effector state, theme changes by configprovider
all
content in app moved to ant-design layout

META DATA: issue13

makhnatkin commented on src/App.tsx:
```diff
@@ -1,39 +1,57 @@
-    </div>
+    <ConfigProvider
+      theme={{
+        algorithm: themeValue === "dark" ? theme.darkAlgorithm : theme.defaultAlgorithm,
```
use themes.DARK inside "dark"

Hellomyguest replied:
themeValue values changed to "DARK" and "LIGHT"

makhnatkin commented on src/features/theme-switcher/ui/ThemeSwitcher.tsx:
```diff
@@ -0,0 +1,19 @@
+  };
+
+  return (
+    <Space style={{color: '#fff'}}>
```
1) use css variables instead of '#fff'
2) use classmanes with css in js instead of style attr

makhnatkin replied:
I think you should use Space without color

Hellomyguest replied:
css variables added
used classnames instead of inline styles
By default text in Space color is black, so it will not be visible in light theme, cause the header is also black.

makhnatkin commented on src/features/theme-switcher/ui/ThemeSwitcher.tsx:
```diff
@@ -0,0 +1,19 @@
+  return (
+    <Space style={{color: '#fff'}}>
+      День
+      <Switch checked={themeValue === "dark"} onChange={onChange} style={{backgroundColor: '#d9d9d9'}}/>
```
1. use css variables instead of '#fff'
2. use classmanes with css in js instead of style attr

makhnatkin replied:
I think you should use Switch without color

Commit:
feat(components): make required fixes of ThemeSwitcher

use className beside inline css
change theme variable
add className naming rules to readme

META DATA: issue13

Commit:
feat(components): initially theme is set according to user preferences

META DATA: issue13

Commit:
feat(components): fix default theme value

META DATA: issue13

Commit:
feat(components): add PageHeader component

META DATA: issue13

makhnatkin commented on README.md:
````diff
@@ -1,24 +1,76 @@
+[![StackShare](https://img.shields.io/badge/tech-stack-0690fa.svg?style=flat)](https://stackshare.io/stackshare/awesome-stacks)
 
-```javascript
+- #### Development environment: <br>**Vite** (https://vitejs.dev/)
````
update with merge main to this branch

Hellomyguest replied:
done

makhnatkin commented on README.md:
````diff
@@ -1,24 +1,76 @@
+
+```css
+/* ListingCard.css */
+.ListingCard { }
````
Why?

Hellomyguest replied:
According to [Airbnb CSS/ SASS styleguide](https://github.com/airbnb/css#oocss-and-bem)

makhnatkin commented on src/App.tsx:
```diff
@@ -1,58 +1,30 @@
 
 const { Content, Header } = Layout;
-type Theme = 'dark' | 'light';
+type Theme = "DARK" | "LIGHT";
```
type Theme = "dark" | "light";

Hellomyguest replied:
done

makhnatkin commented on src/App.tsx:
```diff
@@ -1,58 +1,30 @@
       theme={{
-        algorithm: themeValue === "dark" ? theme.darkAlgorithm : theme.defaultAlgorithm,
+        algorithm:
+          themeValue === "DARK" ? theme.darkAlgorithm : theme.defaultAlgorithm,
```
themeValue === themes.DARK ? theme.darkAlgorithm : theme.defaultAlgorithm,

Hellomyguest replied:
done

makhnatkin commented on src/App.tsx:
```diff
@@ -1,58 +1,30 @@
-        <Header
-          style={{ display: "flex", justifyContent: "flex-end" }}
-        >
+        <Header className="Header">
```
Why do we need a cool one here? what does it give?

Hellomyguest replied:
I don't understand what cool here means.
This App.tsx is outdated, in last commit added custom PageHeader component

makhnatkin commented on src/App.tsx:
```diff
@@ -1,58 +1,30 @@
 };
 
-export default App;
+export default App;
```
dont use export default
plz add eslint rule for this

Hellomyguest replied:
done

makhnatkin commented on src/features/theme-switcher/ui/ThemeSwitcher.css:
```diff
@@ -0,0 +1,7 @@
+.Space {
```
Is this consistent with the AntD documentation?

Hellomyguest replied:
Ant design for styling space use inline styles, we agreed to use className instead of inline styles.
Example from ant-design: 
![image](https://user-images.githubusercontent.com/48411208/206223182-eddc1e3f-fe64-471e-baf6-b4dbb08403b7.png)

makhnatkin commented on src/features/theme-switcher/ui/ThemeSwitcher.tsx:
```diff
@@ -1,18 +1,19 @@
 
   const onChange = (checked: boolean) => {
-    checked ? themeChangedFn("dark") : themeChangedFn("light");
+    checked ? themeChangedFn("DARK") : themeChangedFn("LIGHT");
```
checked ? themeChangedFn(themes.DARK) : themeChangedFn(themes.LIGHT);

makhnatkin replied:
onChange should be replaced by handleChange
also use effector.prepend for Events https://effector.dev/docs/api/effector/event/#prependfn

It is better to declare handlers outside the component using prepends

Commit:
Merge develop in to the branch

Commit:
feat(components): make required fixes

META DATA: issue13

Commit:
Merge develop in this branch

Commit:
refactor(custom): disable eslint rule import/prefer-default-export

META DATA: issue13

Commit:
fix(components): themeSwitcher change onChange to handleChange

META DATA: issue13

makhnatkin commented on src/features/theme-switcher/model.ts:
```diff
@@ -0,0 +1,8 @@
+import { createEvent, createStore } from "effector";
+
+const defaultDark = window.matchMedia("(prefers-color-scheme: dark)").matches;
+export const $theme = createStore(defaultDark? "dark" : "light");
```
nit
in my opition we should use enums here
sorry, best way will be use enums
```
export enum Themes {
    DEFAULT = 'default',
    DARK = 'dark'
}
```

Hellomyguest replied:
done

Commit:
perf(components): themeSwitcher: handleChange moved outside component

ThemeSwitcher: handle function moved outside component using prepend
add enum Themes

META DATA: issue13
